Bridge the ide/api.lsp services for ide/lsp.client

lahodaj opened the pull request:
Discussed before, if a module provides implements `ide/api.lsp` services, it might be good to use them internally in the IDE. This is a bit more polished prototype doing that. It is done by having an internal LS server bridging the services to `lsp.client`. Note that the data are not serialized, they are just converted from the `ide/api.lsp` structures to the LS structures, and then to the IDE structures.

CC @jtulach

jtulach commented:
@lahodaj thank you. I can confirm your patch does work. It **displays errors/hints** and **shows code completion**. All of that with just a three line change in Enso `layer.xml`:

![NetBeans+PR7811+Enso](https://github.com/user-attachments/assets/23d632fb-c24e-41f1-a78b-61786c6a9cc3)

However **navigator is broken**. Without your PR the structure is visible (thanks to #7483), but with your PR it disappears...

lahodaj commented:
I've added bridging for `documentSymbols`, which should fix the navigator (and should also give you breadcrumbs for free, I think). I think this issue with the navigator demonstrates the existing approach is quite fragile.

I've also moved the annotation to `api.lsp`, as requested elsewhere.

jtulach commented:
Enso works with the current state of the PR:
- errors
- code completion
- navigator

I don't see breadcrumbs, but that OK. Please get this into a release!
